cli/src/dap_client.rs

```diff
@@ -0,0 +1,88 @@
+use std::str::FromStr;
+use url::Url;
+
+use janus_client;
+
+use janus_messages::{Duration, TaskId};
+use prio::vdaf::prio3::Prio3Histogram;
+
+use crate::{CliResult, DetermineAccountId, Error, Output};
+use clap::Subcommand;
+use divviup_client::{self, Histogram};
+
+#[derive(Subcommand, Debug)]
+pub enum DapClientAction {
+    /// create a new task for the target account
+    Upload {
+        #[arg(long)]
+        task_id: String,
+        #[arg(long)]
+        value: usize,
+    },
+}
+
+impl DapClientAction {
+    pub(crate) async fn run(
+        self,
+        account_id: DetermineAccountId,
+        client: divviup_client::DivviupClient,
+        _output: Output,
+    ) -> CliResult {
+        let account_id = account_id.await?;
+
+        match self {
+            DapClientAction::Upload { task_id, value } => {
+                let task = client.task(&task_id).await?;
+
+                let aggregators = client.aggregators(account_id).await?;
+
+                let mut leader_url: Option<Url> = None;
+                let mut helper_url: Option<Url> = None;
+
+                for a in &aggregators {
+                    if a.id == task.leader_aggregator_id {
+                        leader_url = Some(a.dap_url.clone());
+                    }
+                    if a.id == task.helper_aggregator_id {
+                        helper_url = Some(a.dap_url.clone());
+                    }
+                }
+                if leader_url.is_none() {
+                    return Err(Error::Other("leader URL unset".into()));
+                }
+                if helper_url.is_none() {
+                    return Err(Error::Other("leader URL unset".into()));
+                }
```

Comment:
Maybe we need a helper function in the client for this, or not return aggregators as a `Vec<Aggregator>`, since this isn't the first time we've had to write this construct https://github.com/divviup/sre-tools/blob/main/janus-load-generator/src/divviup.rs#L49.

(nothing actionable for you, this LGTM.)